File: 0890-lemonade-change/0890-lemonade-change.go
func lemonadeChange(bills []int) bool {
    five, ten := 0, 0

    for _, bill := range bills {
        if bill == 5 {
            five++
        } else if bill == 10 {
            if five > 0 {
                five--
                ten++
            } else {
                return false
            }
        } else { // bill == 20
            if ten > 0 && five > 0 {
                ten--
                five--
            } else if five >= 3 {
                five -= 3
            } else {
                return false
            }
        }
    }
    return true
}

File: 0890-lemonade-change/0890-lemonade-change.cpp
class Solution {
public:
    bool lemonadeChange(vector<int>& bills) {
        if (bills[0] > 5)
            return false;
        int five = 0, ten = 0;

        for (const auto& b : bills) {
            if (b == 5)
                five++;
            else if (b == 10) {
                if (five > 0) {
                    five--;
                    ten++;
                } else
                    return false;
            } else {
                if (five > 0 && ten > 0)
                    five--,ten--;
                else if (five >= 3)
                    five -= 3;
                else
                    return false;
            }
        }
        return true;
    }
};